scripts/install_deps.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+GLFW_URL="https://github.com/glfw/glfw/archive/refs/tags/3.4.tar.gz"
+GLEW_URL="https://github.com/nigels-com/glew/releases/download/glew-2.2.0/glew-2.2.0.tgz"
+ASSIMP_URL="https://github.com/assimp/assimp/archive/refs/tags/v5.4.0.tar.gz"
+
+
+LIB_PREFIX=~/libs
+ARCHIVE="sources"
+
+clear () {
+	rm -rf ~/.tmp_deps
+	mkdir ~/.tmp_deps
+	cd ~/.tmp_deps
+}
+
+cmake_install () {
+	cmake .
+	cmake --build .
+	cmake --install . --prefix "$LIB_PREFIX"
+}
+
+download_lib () {
+	wget $1 -O "$ARCHIVE"
+	tar xf "$ARCHIVE"
+	rm -rf "$ARCHIVE"	
+}
+
+clear
+download_lib "$GLFW_URL"
+cd *
+cmake_install
+
+clear
+download_lib "$GLEW_URL"
+cd */build/cmake
+cmake_install
+
+clear
+download_lib "$ASSIMP_URL"
+cd *
+cmake_install
+
+clear
```

Comment:
How about we make this code a little bit better? Something like:
```bash

LIB_PREFIX=~/libs
ARCHIVE="sources"
TMP_DIR=~/.tmp_deps

declare -A LIBRARIES=( ["glfw"]="$GLFW_URL" ["glew/build/cmake"]="$GLEW_URL" ["assimp"]="$ASSIMP_URL")

cleanup() {
    if [[ -d "${TMP_DIR}" ]]; then
        rm -rf "${TMP_DIR}"
    fi
    mkdir -p "${TMP_DIR}"
}

cmake_install() {
	local dir=$1
	cd ${TMP_DIR}/${dir}
	cmake .
	cmake --build .
	cmake --install . --prefix "$LIB_PREFIX"
}

download_and_extract_lib() {
	local url=$1
    cd ${TMP_DIR}
	wget "${url}" -O "${ARCHIVE}"
	tar xf "${ARCHIVE}"
	rm -f "${ARCHIVE}"
}

for lib in "${!LIBRARIES[@]}"; do
	cleanup
	download_and_extract_lib "${LIBRARIES[$lib]}"
	cmake_install $lib
done

cleanup
```

Reply:
Adding a suggestion which can be merged instantly (however I only tested that on MacOS)

```suggestion
#!/bin/bash

GLFW_URL="https://github.com/glfw/glfw/archive/refs/tags/3.4.tar.gz"
GLEW_URL="https://github.com/nigels-com/glew/releases/download/glew-2.2.0/glew-2.2.0.tgz"
ASSIMP_URL="https://github.com/assimp/assimp/archive/refs/tags/v5.4.0.tar.gz"


LIB_PREFIX=~/libs
ARCHIVE="sources"
TMP_DIR=~/.tmp_deps

clear () {
	rm -rf $TMP_DIR
	mkdir $TMP_DIR
	cd $TMP_DIR
}

cmake_install () {
	cmake .
	cmake --build .
	cmake --install . --prefix "$LIB_PREFIX"
}

download_lib () {
	wget $1 -O "$ARCHIVE"
	tar xf "$ARCHIVE"
	rm -rf "$ARCHIVE"	
}

urls=("$GLFW_URL" "$GLEW_URL" "$ASSIMP_URL")

for url in "${urls[@]}"; do
  clear
  download_lib "$url"
  cd *
  cmake_install
  cd -
done

clear
```